packages/@aws-cdk-containers/ecs-service-extensions/test/container.test.ts

```diff
@@ -0,0 +1,315 @@
+          'Arn',
+        ],
+      },
+    });
+
+
+  });
+
+  test('should be able to add user-provided log group in the log driver options', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    stack.node.setContext(cxapi.ECS_SERVICE_EXTENSIONS_ENABLE_DEFAULT_LOG_DRIVER, true);
+
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new ecs.Cluster(stack, 'Cluster', { vpc });
+    cluster.addAsgCapacityProvider(new ecs.AsgCapacityProvider(stack, 'Provider', {
+      autoScalingGroup: new autoscaling.AutoScalingGroup(stack, 'DefaultAutoScalingGroup', {
+        vpc,
+        machineImage: ec2.MachineImage.latestAmazonLinux(),
+        instanceType: new ec2.InstanceType('t2.micro'),
+      }),
+    }));
+
+    const environment = new Environment(stack, 'production', {
+      vpc,
+      cluster,
+      capacityType: EnvironmentCapacityType.EC2,
+    });
+    const serviceDescription = new ServiceDescription();
+    const taskRole = new iam.Role(stack, 'CustomTaskRole', {
+      assumedBy: new iam.ServicePrincipal('ecs-tasks.amazonaws.com'),
+    });
+
+    serviceDescription.add(new Container({
+      cpu: 256,
+      memoryMiB: 512,
+      trafficPort: 80,
+      image: ecs.ContainerImage.fromRegistry('nathanpeck/name'),
+      logGroup: new awslogs.LogGroup(stack, 'MyLogGroup'),
+    }));
+
+    new Service(stack, 'my-service', {
+      environment,
+      serviceDescription,
+      taskRole,
+    });
+
+    // THEN
+    expect(stack).toCountResources('AWS::ECS::Service', 1);
+
+    // Ensure that the log group was created
+    expect(stack).toHaveResource('AWS::Logs::LogGroup');
+
+    expect(stack).toHaveResource('AWS::ECS::TaskDefinition', {
+      ContainerDefinitions: [
+        {
+          Cpu: 256,
+          Essential: true,
+          Image: 'nathanpeck/name',
+          LogConfiguration: {
+            LogDriver: 'awslogs',
+            Options: {
+              'awslogs-group': {
+                Ref: 'MyLogGroup5C0DAD85',
+              },
+              'awslogs-stream-prefix': 'my-service',
+              'awslogs-region': {
+                Ref: 'AWS::Region',
+              },
+            },
+          },
+          Memory: 512,
+          Name: 'app',
+          PortMappings: [
+            {
+              ContainerPort: 80,
+              Protocol: 'tcp',
+            },
+          ],
+          Ulimits: [
+            {
+              HardLimit: 1024000,
+              Name: 'nofile',
+              SoftLimit: 1024000,
+            },
+          ],
+        },
+      ],
+      Cpu: '256',
+      Family: 'myservicetaskdefinition',
+      Memory: '512',
+      NetworkMode: 'awsvpc',
+      RequiresCompatibilities: [
+        'EC2',
+        'FARGATE',
+      ],
+      TaskRoleArn: {
+        'Fn::GetAtt': [
+          'CustomTaskRole3C6B13FD',
+          'Arn',
+        ],
+      },
+    });
+
+
+  });
+
+  test('should error when log group is provided but feature flag is not set', () => {
```

Comment:
Please add unit tests for the other error scenario as well - a logGroup is provided on the Container extension, but also a logging extension is added to the service description.

Reply:
oh, nevermind. I see that was in another unit test file.